### PR TITLE
remove validation GCE-ism

### LINKED ISSUE #114436: validation has a GCE-ism baked in
pkg/apis/core/validation/validation.go

```
func ValidateReadOnlyPersistentDisks(volumes []core.Volume, fldPath *field.Path) field.ErrorList {                                                                             
    allErrs := field.ErrorList{}
    for i := range volumes {
        vol := &volumes[i]
        idxPath := fldPath.Index(i)
        if vol.GCEPersistentDisk != nil {
            if !vol.GCEPersistentDisk.ReadOnly {
                allErrs = append(allErrs, field.Invalid(idxPath.Child("gcePersistentDisk", "readOnly"), false, "must be true for replicated pods > 1; GCE PD can only be mounted on multiple machines if it is read-only"))
            }    
        }    
        // TODO: What to do for AWS?  It doesn't support replicas
    }    
    return allErrs
}
```

We should probably just get rid of this.

### PR DESCRIPTION
**Pull request by mfordjody**
#### What type of PR is this?
/kind cleanup
/kind api-change

#### What this PR does / why we need it:
remove GCE-ism validation

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/114436

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by mfordjody** [comment]
This pull request source https://github.com/kubernetes/kubernetes/pull/116142 and https://github.com/kubernetes/kubernetes/pull/114631

File: pkg/apis/core/validation/validation.go [hunk 1]
```diff
@@ -5319,12 +5324,17 @@ func ValidatePodTemplateSpecForRC(template *core.PodTemplateSpec, selectorMap ma
...
 // ValidateReplicationControllerSpec tests if required fields in the replication controller spec are set.
-func ValidateReplicationControllerSpec(spec *core.ReplicationControllerSpec, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+func ValidateReplicationControllerSpec(spec, oldSpec *core.ReplicationControllerSpec, fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, ValidateNonnegativeField(int64(spec.MinReadySeconds), fldPath.Child("minReadySeconds"))...)
 	allErrs = append(allErrs, ValidateNonEmptySelector(spec.Selector, fldPath.Child("selector"))...)
 	allErrs = append(allErrs, ValidateNonnegativeField(int64(spec.Replicas), fldPath.Child("replicas"))...)
-	allErrs = append(allErrs, ValidatePodTemplateSpecForRC(spec.Template, spec.Selector, spec.Replicas, fldPath.Child("template"), opts)...)
+	// oldSpec is not empty, pass oldSpec.template.
+	var oldTemplates *core.PodTemplateSpec
```
Comment (thockin) [inline pkg/apis/core/validation/validation.go:5333]: `oldTemplate` - no "s"

File: pkg/apis/core/validation/validation.go [hunk 2]
```diff
@@ -5344,17 +5354,29 @@ func ValidatePodTemplateSpec(spec *core.PodTemplateSpec, fldPath *field.Path, op
...
+		return field.ErrorList{}
+	}
+
+	isWriteablePD := func(vol *core.Volume) bool {
+		return vol.GCEPersistentDisk != nil && !vol.GCEPersistentDisk.ReadOnly
+	}
+
+	for i := range oldVolumes {
+		if isWriteablePD(&oldVolumes[i]) {
+			return allErrs
```
Comment (thockin) [inline pkg/apis/core/validation/validation.go:5371]: for clarity, `return field.ErrorList{}`

File: pkg/apis/core/validation/validation.go [hunk 3]
```diff
@@ -5344,17 +5354,29 @@ func ValidatePodTemplateSpec(spec *core.PodTemplateSpec, fldPath *field.Path, op
 	return allErrs
 }
 
-func ValidateReadOnlyPersistentDisks(volumes []core.Volume, fldPath *field.Path) field.ErrorList {
+// ValidateReadOnlyPersistentDisks validates pd reads and writes
+func ValidateReadOnlyPersistentDisks(volumes, oldVolumes []core.Volume, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
```
Comment (thockin) [inline pkg/apis/core/validation/validation.go:5359]: stick this AFTER the short-circuit checks

File: pkg/apis/apps/validation/validation_test.go [hunk 4]
```diff
@@ -2875,13 +3120,14 @@ func TestValidateReplicaSetUpdate(t *testing.T) {
 			update: apps.ReplicaSet{
 				ObjectMeta: metav1.ObjectMeta{Name: "abc", Namespace: metav1.NamespaceDefault},
 				Spec: apps.ReplicaSetSpec{
-					Replicas: 3,
+					Replicas: 1,
```
Comment (thockin) [inline pkg/apis/apps/validation/validation_test.go:3123]: this should stay 3 because 1 is the default, so this update is a no-op.

**Review by thockin — COMMENTED** [review]
Almost there!

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**